File: app/src/main/java/io/ipoli/android/app/ui/formatters/TimerFormatter.java
package io.ipoli.android.app.ui.formatters;

import java.util.concurrent.TimeUnit;

/**
 * Created by Polina Zhelyazkova <polina@ipoli.io>
 * on 1/28/16.
 */
public class TimerFormatter {

    public static String format(long timerMillis) {
        int hours = (int) TimeUnit.MILLISECONDS.toHours(timerMillis);
        int minutes = (int) TimeUnit.MILLISECONDS.toMinutes(timerMillis) - hours * 60;
        int seconds = (int) TimeUnit.MILLISECONDS.toSeconds(timerMillis) - hours * 3600 - minutes * 60;

        String text = String.format("%02d:%02d", minutes, seconds);
        if(hours > 0) {
            text = String.format("%d:", hours) + text;
        }

        return text;
    }
}

File: app/src/main/java/io/ipoli/android/quest/timer/view/formatter/TimerFormatter.kt
package io.ipoli.android.quest.timer.view.formatter

import java.util.concurrent.TimeUnit

/**
 * Created by Venelin Valkov <venelin@io.ipoli.io>
 * on 01/17/2018.
 */
object TimerFormatter {

    fun format(timerMillis: Long): String {
        val hours = TimeUnit.MILLISECONDS.toHours(timerMillis).toInt()
        val minutes = TimeUnit.MILLISECONDS.toMinutes(timerMillis).toInt() - hours * 60
        val seconds =
            TimeUnit.MILLISECONDS.toSeconds(timerMillis).toInt() - hours * 3600 - minutes * 60

        var text = String.format("%02d:%02d", minutes, seconds)
        if (hours > 0) {
            text = String.format("%d:", hours) + text
        }

        return text
    }
}